staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -31,15 +31,59 @@ import (
 	"k8s.io/klog"
 )
 
-// SharedInformer has a shared data cache and is capable of distributing notifications for changes
-// to the cache to multiple listeners who registered via AddEventHandler. If you use this, there is
-// one behavior change compared to a standard Informer.  When you receive a notification, the cache
-// will be AT LEAST as fresh as the notification, but it MAY be more fresh.  You should NOT depend
-// on the contents of the cache exactly matching the notification you've received in handler
-// functions.  If there was a create, followed by a delete, the cache may NOT have your item.  This
-// has advantages over the broadcaster since it allows us to share a common cache across many
-// controllers. Extending the broadcaster would have required us keep duplicate caches for each
-// watch.
+// SharedInformer provides eventually consistent linkage of its
+// clients to the authoritative state of a given collection of
+// objects.  An object is identified by its API group, kind/resource,
+// namespace, and name.  One SharedInfomer provides linkage to objects
+// of a particular API group and kind/resource, and may be further
+// restricted to one namespace and/or by label selector and/or field
```

Comment:
Is the restriction global, per-client, or both?

Reply:
This is in the contex of a description of the SharedInformer as a whole, before clients are even discussed.  Does that answer the question?

Reply:
Yes--but I think it would be better to be explicit in the text.

Reply:
fixed